WNPRC_Purchasing/src/client/components/LineItemsPanel.tsx

```diff
@@ -50,67 +53,94 @@ export const LineItemsPanel: FC<Props> = memo(props => {
-                                    : accumulatedVal;
-                            }, 0)
-                        )}
-                    </Col>
-                    <Col xs={1}></Col>
-                </Row>
-            </div>
-            <div className="add-item">
-                <span id="add-line-item-row" title="Add item" className="add-item-icon" onClick={onClickAddRow}>
-                    <FontAwesomeIcon className="fa-faPlusCircle" icon={faPlusCircle} color="green" /> Add item
-                </span>
-            </div>
-            {errorMsg && <div className="alert alert-danger">{errorMsg}</div>}
-        </Panel>
+                    <div>
+                        {
+                            <Row className="line-item-row-header">
+                                <Col xs={4}>Part no./Item description *</Col>
+                                <Col xs={1}>Unit *</Col>
+                                { (isAdmin || !canUpdate) &&
+                                    <Col xs={1}>Unit Cost ($) *</Col>
+                                }
+                                <Col xs={1}>Quantity *</Col>
+                                { (isAdmin || !canUpdate) &&
+                                    <Col xs={1}>Subtotal ($)</Col>
+                                }
+                                <Col xs={!hasRequestId ? 2 : 1}>Controlled substance</Col>
+                                { hasRequestId && (isAdmin || canUpdate) &&
+                                    <Col xs={1}>Quantity received</Col>
+                                }
+                            </Row>
+                        }
+                    </div>
+                    <div>
+                        {lineItems.map((lineItem, idx) => {
+                            return (
+                                <LineItemRow
+                                    rowIndex={idx}
+                                    key={'line-item-' + idx}
+                                    model={lineItem}
+                                    onInputChange={lineItemRowChange}
+                                    onDelete={onDeleteRow}
+                                    isAdmin={isAdmin}
+                                    canUpdate={canUpdate}
+                                    canInsert={canInsert}
+                                    hasRequestId={hasRequestId}
+                                />
+                            );
+                        })}
+                    </div>
+                    <div>
+                        {
+                            (isAdmin || !canUpdate) && (
+                                <>
+                                    <div>
+                                        <Row>
+                                            <Col xs={4}></Col>
+                                            <Col xs={1}></Col>
+                                            <Col xs={1}></Col>
```

Comment:
This is kind of strange.  Can you not just do `<Col xs={6} />`?  There are other places doing this as well that at least should be collapsed to empty tag.

Reply:
was trying to match the header cols, but yeah can use 6, now updated